django/contrib/postgres/search.py

```diff
@@ -167,50 +167,35 @@ class SearchQuery(SearchQueryCombinable, Value):
     }
 
     def __init__(self, value, output_field=None, *, config=None, invert=False, search_type='plain'):
-        self.config = SearchConfig.from_parameter(config)
-        self.invert = invert
-        if search_type not in self.SEARCH_TYPES:
+        function = self.SEARCH_TYPES.get(search_type)
+        if function is None:
             raise ValueError("Unknown search_type argument '%s'." % search_type)
-        self.search_type = search_type
-        super().__init__(value, output_field=output_field)
-
-    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
-        resolved = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
-        if self.config:
-            resolved.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
-        return resolved
-
-    def as_sql(self, compiler, connection):
-        params = [self.value]
-        function = self.SEARCH_TYPES[self.search_type]
-        if self.config:
-            config_sql, config_params = compiler.compile(self.config)
-            template = '{}({}, %s)'.format(function, config_sql)
-            params = config_params + [self.value]
-        else:
-            template = '{}(%s)'.format(function)
-        if self.invert:
-            template = '!!({})'.format(template)
-        return template, params
-
-    def _combine(self, other, connector, reversed):
-        combined = super()._combine(other, connector, reversed)
-        combined.output_field = SearchQueryField()
-        return combined
+        if not hasattr(value, 'resolve_expression'):
+           value = Value(value)
+        expressions = (value,)
+        if config is not None:
+            config = SearchConfig.from_parameter(config)
+            expressions = (config,) + expressions
+        super().__init__(*expressions, output_field=output_field, function=function, invert=invert)
```

Comment:
We can keep `invert` directly in the `SearchQuery` instead of `extra`, e.g.
```python
...
self.invert = invert
super().__init__(*expressions, output_field=output_field, function=function)
```